homeassistant/components/meteoswiss/coordinator.py

```diff
@@ -0,0 +1,54 @@
+"""DataUpdateCoordinator for Met.no integration."""
+
+import dataclasses
+import datetime
+import logging
+
+import meteoswiss_async
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, POSTAL_CODE, POSTAL_CODE_ADDITIONAL_NUMBER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(kw_only=True, frozen=True)
+class MeteoSwissData:
+    """Data that gets updated by the coordinator."""
+
+    weather: meteoswiss_async.Weather
+
+
+class MeteoSwissDataUpdateCoordinator(DataUpdateCoordinator[MeteoSwissData]):
+    """Class to manage fetching Met data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api_client: meteoswiss_async.MeteoSwissClient,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize global Met data updater."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=datetime.timedelta(minutes=5),
+        )
+        self._api_client = api_client
+        self._postal_code: str = str(config_entry.data[POSTAL_CODE])
+        self._additional_number: int = config_entry.data[POSTAL_CODE_ADDITIONAL_NUMBER]
```

Comment:
```suggestion
    """Class to manage fetching Met data."""

    config_entry: ConfigEntry

    def __init__(
        self,
        hass: HomeAssistant,
        api_client: meteoswiss_async.MeteoSwissClient,
    ) -> None:
        """Initialize global Met data updater."""
        super().__init__(
            hass,
            _LOGGER,
            name=DOMAIN,
            update_interval=datetime.timedelta(minutes=5),
        )
        self._api_client = api_client
        self._postal_code: str = str(self.config_entry.data[POSTAL_CODE])
        self._additional_number: int = self.config_entry.data[POSTAL_CODE_ADDITIONAL_NUMBER]
```

Reply:
I don't understand the proposed change here to move the config entry as a class variable. 

A config entry should be a property of the Coordinator instance (each coordinator will have a reference to a different config entry) therefore does not make sense to define it as a class variable. More information about the recommendations in Python can be found in: https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables

Reply:
The constructor of the coordinator fetches the config entry automatically, so technically there's no need to ask it as parameter

Reply:
Done